superset/assets/src/dashboard/actions/dashboardState.js

```diff
@@ -134,19 +135,18 @@ export function saveDashboardRequestSuccess() {
 export function saveDashboardRequest(data, id, saveType) {
   const path = saveType === SAVE_TYPE_OVERWRITE ? 'save_dash' : 'copy_dash';
 
-  return dispatch =>
-    SupersetClient.post({
+  return dispatch => {
+    dispatch({ type: UPDATE_COMPONENTS_PARENTS_LIST });
```

Comment:
if this is a `Promise` you should be able to chain it with the `POST` request. Is there a reason not to do that?

Reply:
`dispatch` is synchronous function, the dispatched action can be asynchronous, like returning a promise, so that we can chain actions after it. 
https://stackoverflow.com/questions/43276291/is-store-dispatch-in-redux-synchronous-or-asynchronous

The action `{ type: UPDATE_COMPONENTS_PARENTS_LIST }` is synchronous. Browser will always run this action first, then do post. (so that no need to chain in Promise object)